app/questionnaire/navigator.py

```diff
@@ -116,6 +110,17 @@ def build_path(self, blocks, this_location):
 
         return path
 
+    def _remove_rule_answers(self, rule, location):
+        # We're jumping backwards, so need to delete all answers from which
+        # route is derived
+        if not SchemaHelper.is_goto_meta_rule(rule):
+            for condition in rule['goto']['when']:
+                self.answer_store.remove(answer_id=condition['id'],
+                                         answer_instance=0,
```

Comment:
I assume we don't support routing based on anything other than the first answer_instance right now? If that changes in future this could fail... not sure what you can do now about that though, just highlighting it.